src/Common/src/System/Drawing/ColorTable.cs

```diff
@@ -0,0 +1,87 @@
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace System.Drawing
+{
+    internal static class ColorTable
+    {
+        private static string ColorConstantsLock = "colorConstants";
+        private static Dictionary<string, Color> colorConstants;
+        private static string SystemColorConstantsLock = "systemColorConstants";
+        private static Dictionary<string, Color> systemColorConstants;
+
+        internal static Dictionary<string, Color> Colors
+        {
+            get
+            {
+                if (colorConstants == null)
+                {
+                    lock (ColorConstantsLock)
+                    {
+                        if (colorConstants == null)
+                        {
+                            Dictionary<string, Color> tempHash = new Dictionary<string, Color>(StringComparer.OrdinalIgnoreCase);
+                            FillConstants(tempHash, typeof(Color));
+                            colorConstants = tempHash;
+                        }
+                    }
+                }
+
+                return colorConstants;
+            }
+        }
+
+        internal static Dictionary<string, Color> SystemColors
+        {
+            get
+            {
+                if (systemColorConstants == null)
+                {
+                    lock (SystemColorConstantsLock)
+                    {
+                        if (systemColorConstants == null)
+                        {
+                            Dictionary<string, Color> tempHash = new Dictionary<string, Color>(StringComparer.OrdinalIgnoreCase);
+                            FillConstants(tempHash, typeof(System.Drawing.SystemColors));
+                            systemColorConstants = tempHash;
+                        }
+                    }
+                }
+
+                return systemColorConstants;
+            }
+        }
+
+        private static void FillConstants(Dictionary<string, Color> colors, Type enumType)
+        {
+            MethodAttributes attrs = MethodAttributes.Public | MethodAttributes.Static;
+            PropertyInfo[] props = enumType.GetProperties();
+
+            for (int i = 0; i < props.Length; i++)
+            {
+                PropertyInfo prop = props[i];
+                if (prop.PropertyType == typeof(Color))
+                {
+                    MethodInfo method = prop.GetGetMethod();
+                    if (method != null && (method.Attributes & attrs) == attrs)
+                    {
+                        colors[prop.Name] = (Color) prop.GetValue(null, null);
+                    }
+                }
+            }
+        }
+
+        internal static Color? GetNamedColor(string name)
```

Comment:
Given then usage, it might be better if this were TryGetNamedColor.  Then the implementation could also be even more concise:
```C#
internal static bool TryGetNamedColor(string name, out Color result) =>
    Colors.TryGetValue(name, out result) || SystemColors.TryGetValue(name, out result);
```